pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -581,7 +581,7 @@ func (b *BridgeBindMechanism) learnInterfaceRoutes() error {
 }
 
 func (b *BridgeBindMechanism) decorateDhcpConfigRoutes(dhcpConfig *cache.DhcpConfig) {
-	dhcpConfig.AdvertisingIPAddr = b.routes[0].Gw
+	dhcpConfig.Gateway = b.routes[0].Gw
```

Comment:
This is just a question, but I always think of this when I read it.
Shouldn't this method return an error? It takes index from slice, which may not be there.
I know we're checking the slice contains at least one record before calling the method.
It's just if somebody ever wanted to call it elsewhere, it's easy to forget to check that.

I'm just interested what's your opinion.

Reply:
I want to push a follow up where we "redefine" the concept of default route. This - grabbing the first route - sucks (imho).

I've spent almost 2 days debugging a scenario with secondary networks (and blaming the code) because I just didn't remember that multus is unable to touch the pod's routes unless via an annotation in the pod.

I think that ultimately you're right, but I'd go a step further; I think we should: 
  - come up with a better definition of default route. I think the best alternative is to replicate netlink's definition of it (since our routes are https://github.com/vishvananda/netlink/blob/f9d8f911dbf94ecc5e2bf00ebc45064b63d8d60d/route.go#L34). It would be something like: https://github.com/vishvananda/netlink/blob/f9d8f911dbf94ecc5e2bf00ebc45064b63d8d60d/route_test.go#L765
  - possibly return an error. The caller would just log, and drop the error, since we can have scenarios where pods don't have a default route (a single multus network, not marked as default).

Since this is a "touchy" subject, I'd rather tackle it in a follow up. By touchy, I mean that the tricky thing is defining what the behavior will be.

Reply:
>- come up with a better definition of default route. I think the best alternative is to replicate netlink's definition of it (since our routes are https://github.com/vishvananda/netlink/blob/f9d8f911dbf94ecc5e2bf00ebc45064b63d8d60d/route.go#L34). It would be something like: https://github.com/vishvananda/netlink/blob/f9d8f911dbf94ecc5e2bf00ebc45064b63d8d60d/route_test.go#L765
>- possibly return an error. The caller would just log, and drop the error, since we can have scenarios where pods don't have a default route (a single multus network, not marked as default).

Thanks for elaborating on this, I agree with both points